homeassistant/components/ecoal_boiler.py

```diff
@@ -21,31 +22,77 @@
 DEFAULT_USERNAME = "admin"
 DEFAULT_PASSWORD = "admin"
 
+
+# Available pump ids
+# Available as switches
+PUMP_IDNAMES = (
+    "central_heating_pump",
+    "domestic_hot_water_pump",
+    "central_heating_pump2",
+)
+
+# Available temp sensor ids
+# Available as sensors
+SENSOR_IDS = (
+    "outdoor_temp",
+    "indoor_temp",
+    "indoor2_temp",
+    "domestic_hot_water_temp",
+    "target_domestic_hot_water_temp",
+    "feedwater_in_temp",
+    "feedwater_out_temp",
+    "target_feedwater_temp",
+    "coal_feeder_temp",
+    "exhaust_temp",
+)
+
+
+CONF_SWITCHES = 'switches'
+SWITCHES_SCHEMA = {}
+for pump_id in PUMP_IDNAMES:
+    SWITCHES_SCHEMA[vol.Optional(pump_id)] = cv.string
```

Comment:
We don't need to allow a custom name per sensor or switch type. Change this validation to a list of types and just check that the configured sensor types are in the list of allowed types.

Default names should be set by the platforms from the sensor or switch types.

Reply:
There is no good default.
Switches as pump can be configured (internally in controller) to perform different tasks depending on heating configuration. 
Sensors can be placed in different places(rooms) so there should be a way to set a name visible in GUI.

Reply:
Names can always be customized. If a unique_id is implemented it can be done very easily.

As long as we can add a separate name for each entity, that will be good enough.

Reply:
OK, so assuming I would like to set name of given pump how that should be done in config?
Component config would be
```
ecoal_boiler:
    switches:
        central_heating_pump
```
where should go name of  _central_heating_pump_  (like _Wall radiators_ ) ?

Reply:
Entity names are now hardcoded.